src/util/migration.rs

```diff
@@ -0,0 +1,31 @@
+use bitcoin_hashes::sha256::Hash as Sha256Hash;
+use bitcoin_hashes::sha256d::Hash as Sha256dHash;
+use bitcoin_hashes::hash160::Hash as Hash160;
+
+use std::io::Write;
+
+pub trait HashFromData {
+	fn from_data(buf: &[u8]) -> Self;
+}
+
+impl HashFromData for Sha256dHash {
+	fn from_data(buf: &[u8]) -> Sha256dHash {
```

Comment:
Hmm, we dont actually use from_data in that many places. Maybe just replace it with Hash::hash? https://docs.rs/bitcoin_hashes/0.3.1/bitcoin_hashes/trait.Hash.html#method.hash

Reply:
done